### PR TITLE
Created ecschema-editing package and moved code.

### PR DESCRIPTION
**Pull request by christophermlawson**
To reduce the size of the ecshema-metadata package, we are moving the schema editing and validation APIs to a new package named 'ecschema-editing'.  We will continue to expand the editing and validation APIs from this location.

### CODE REVIEW

File: docs/changehistory/NextVersion.md [hunk 1]
```diff
@@ -143,3 +145,7 @@ To make it easier to use async APIs while exporting a schema, [IModelExportHandl
 [IModelTransformer.shouldExportSchema]($backend) now gets a [SchemaKey]($ecschema-metadata) schema key as argument, instead of a full [Schema]($ecschema-metadata). If you
 need to check the full schema, return `true` in shouldExportSchema and in [IModelExportHandler.onExportSchema]($backend), you can use the schema object to check and then
 return early.
+
+### [@bentley/imodeljs-backend](https://www.itwinjs.org/reference/ecschema-metadata/)
```
Comment (ColinKerr) [inline docs/changehistory/NextVersion.md:149]: package name is wrong

File: docs/changehistory/NextVersion.md [hunk 2]
```diff
@@ -143,3 +145,7 @@ To make it easier to use async APIs while exporting a schema, [IModelExportHandl
...
+
+To reduce the size and limit the scope of the APIs available in the ecschema-metadata package, all APIs associated with EC Schema editing and validation have been moved to the [@bentley/ecschema-editing](https://www.itwinjs.org/reference/ecschema-editing/) package. This includes all source code under the [Validation](https://www.itwinjs.org/reference/ecschema-metadata/) and [Editing](https://www.itwinjs.org/reference/ecschema-metadata/editing/) folders. All corresponding @beta types defined in the ecschema-metadata package have been deprecated.  All @alpha types have been removed from the ecschema-metadata package.
```
Comment (ColinKerr) [inline docs/changehistory/NextVersion.md:151]: Say that the is the first step in our schema editing story.  Link to the discussions page about it.  

Make a bullet list of what has moved/changed so it's easier to see with a quick glance

**Comment by calebmshafer** [comment]
@christophermlawson I just pushed a change that should fix your issue with copyrights. I had to do a checkout of those specific files from `master` (`git checkout master -- common/scripts/install-rush.js`) after running `rush clean && rush purge && rush install` to avoid the hook messing with it.

You'll probably want to do a `rush clean && rush purge && rush install` to try and fix any local issues with the git hook.

File: core/ecschema-editing/src/Validation/SchemaGraph.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (christophermlawson) [reply core/ecschema-editing/src/Validation/SchemaGraph.ts:1]: @ColinKerr  I'll move it under 'utils' folder.  Export as beta or alpha?  It's marked as internal now.

File: docs/changehistory/NextVersion.md (hunk 2 above)
Comment (christophermlawson) [reply docs/changehistory/NextVersion.md]: There quite a few types.  Not sure if this is what you are looking for...but have a look.

File: core/ecschema-editing/src/Editing/Editor.ts [hunk 4]
```diff
@@ -6,37 +6,76 @@
...
-import { MutableRelationshipClass, RelationshipClass } from "../Metadata/RelationshipClass";
-import { MutableSchema, Schema } from "../Metadata/Schema";
-import { MutableUnit, Unit } from "../Metadata/Unit";
-import { MutableUnitSystem, UnitSystem } from "../Metadata/UnitSystem";
-import { SchemaItemKey, SchemaKey } from "../SchemaKey";
-import { FormatType } from "../utils/FormatEnums";
 import * as Rules from "../Validation/ECRules";
 
+import {
+  AnyEnumerator,
```
Comment (ColinKerr) [inline core/ecschema-editing/src/Editing/Editor.ts:12]: Don't do one import per line, if it's too long for one line just make it a couple of lines

File: core/ecschema-editing/src/Editing/MutableClass.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import {
+  CustomAttribute,
```
Comment (ColinKerr) [inline core/ecschema-editing/src/Editing/MutableClass.ts:6]: do not put one import per line, if needed split into multiple lines

File: core/ecschema-editing/src/Editing/MutableEntityClass.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import {
+  DelayedPromiseWithProps,
```
Comment (ColinKerr) [inline core/ecschema-editing/src/Editing/MutableEntityClass.ts:6]: Same

File: core/ecschema-editing/src/Editing/Editor.ts (hunk 4 above)
Comment (christophermlawson) [reply core/ecschema-editing/src/Editing/Editor.ts:12]: Ok.  I briefly had this import order extension enabled which auto-ordered a couple of files. I will correct.

**Review by ColinKerr — APPROVED** [review]

**Review by pmconne — COMMENTED** [review]
The ecschema-metadata package is included as a "core" package in the [API reference](https://www.itwinjs.org/reference/). Should this package be included there too?

**Comment by christophermlawson** [comment]
Thanks @pmconne, yes it should. I will update the doc.  I also see ecschema-metadata is listed twice, so I'll fix that too.

**Comment by pmconne** [comment]
> Thanks @pmconne, yes it should. I will update the doc. I also see ecschema-metadata is listed twice, so I'll fix that too.

Thanks - please also add it to leftNav.md.

**Comment by christophermlawson** [comment]
@pmconne -  Done.

**Review by pmconne — APPROVED** [review]

File: core/ecschema-editing/package.json [hunk 7]
```diff
@@ -0,0 +1,87 @@
+{
+  "name": "@bentley/ecschema-editing",
+  "version": "2.18.0-dev.2",
+  "description": "ECSchema editing and validation API",
+  "license": "MIT",
+  "main": "lib/ecschema-editing.js",
+  "typings": "lib/ecschema-editing",
+  "imodeljsSharedLibrary": true,
```
Comment (calebmshafer) [inline core/ecschema-editing/package.json:8]: Not needed for a backend only dep

File: core/ecschema-editing/src/Editing/MutableCAClass.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { CustomAttributeClass, CustomAttributeContainerType } from "@bentley/ecschema-metadata";
+
+/**
+ * @internal
```
Comment (calebmshafer) [inline core/ecschema-editing/src/Editing/MutableCAClass.ts:8]: Looks like a bunch of these classes are missing from the api.md file.  Are they not imported into the barrel?

**Review by calebmshafer — CHANGES_REQUESTED** [review]

File: core/ecschema-editing/package.json (hunk 7 above)
Comment (christophermlawson) [reply core/ecschema-editing/package.json]: Done

**Review by calebmshafer — APPROVED** [review]

File: core/ecschema-editing/src/Editing/MutableCAClass.ts (hunk 8 above)
Comment (christophermlawson) [reply core/ecschema-editing/src/Editing/MutableCAClass.ts:8]: No, they are not.  These mutable classes were extracted from the source files in ecschema-metadata (ie. Class.ts) and moved here as individual files.  They are used internally for editing purposes.  What would you like me to do here?

Comment (calebmshafer) [reply core/ecschema-editing/src/Editing/MutableCAClass.ts:8]: Let's just leave them out of the barrel. less chance someone will use them.

**Pull request merged**